bin/sozo/src/commands/execute.rs

```diff
@@ -76,64 +85,76 @@ impl ExecuteArgs {
-                if self.diff {
-                    message.push_str(
-                        " Run the command again with `--diff` to force the fetch of data from the \
-                         chain.",
-                    );
-                }
-                anyhow!(message)
-            })?;
-
-            trace!(
-                contract=?descriptor,
-                entrypoint=self.entrypoint,
-                calldata=?self.calldata,
-                "Executing Execute command."
-            );
-
-            let calldata = if let Some(cd) = self.calldata {
-                calldata_decoder::decode_calldata(&cd)?
-            } else {
-                vec![]
-            };
-
-            let call = Call {
-                calldata,
-                to: contract_address,
-                selector: snutils::get_selector_from_name(&self.entrypoint)?,
-            };
-
             let (provider, _) = self.starknet.provider(profile_config.env.as_ref())?;
 
+            let contracts = utils::contracts_from_manifest_or_diff(
+                self.account.clone(),
+                self.starknet.clone(),
+                self.world,
+                &ws,
+                self.diff,
+            )
+            .await?;
+
             let account = self
                 .account
                 .account(provider, profile_config.env.as_ref(), &self.starknet, &contracts)
                 .await?;
 
-            let invoker = Invoker::new(&account, txn_config);
-            let tx_result = invoker.invoke(call).await?;
+            let mut invoker = Invoker::new(&account, txn_config);
+
+            let mut arg_iter = self.calls.into_iter();
+
+            while let Some(arg) = arg_iter.next() {
+                let tag_or_address = ResourceDescriptor::from_string(&arg)?;
+                let descriptor = tag_or_address.ensure_namespace(&profile_config.namespace.default);
+
+                let contract_address = match &descriptor {
+                    ResourceDescriptor::Address(address) => Some(*address),
+                    ResourceDescriptor::Tag(tag) => contracts.get(tag).map(|c| c.address),
+                    ResourceDescriptor::Name(_) => {
+                        unimplemented!("Expected to be a resolved tag with default namespace.")
+                    }
+                };
+
+                let contract_address = contract_address.ok_or_else(|| {
+                    let mut message = format!("Contract {descriptor} not found in the manifest.");
+                    if self.diff {
+                        message.push_str(
+                            " Run the command again with `--diff` to force the fetch of data from \
+                             the chain.",
+                        );
+                    }
+                    anyhow!(message)
+                })?;
+
+                let entrypoint =
+                    arg_iter.next().ok_or_else(|| anyhow!("Unexpected number of arguments"))?;
```

Comment:
We must be more precise here, because someone running `sozo execute actions` will have this issue, and we can be more informative since we know we're at the `entrypoint` one.